visualize-admin/visualization-tool
# fix: Dimension data type parsing (in cases of two data types)

Fixes #1395.

It looks that it's possible for a cube dimension to define two data types directly, see https://s.zazuko.com/3aCW2sZ (two `sh:datatype` entries).

[The way of parsing dimension data type in cube-view-query-library](https://github.com/zazuko/rdf-cube-view-query/blob/20d3895edbd6c56df59ae04d0291285382304add/lib/CubeDimension.js#L21) was not expecting more than one property (`this.ptr.out(ns.sh.datatype).term` instead of `this.ptr.out(ns.sh.datatype).terms`), which led to data type being undefined.

This on the other hand resulted in a failed check for `shouldLoadMinMaxValues` function that led to loading of every value for numerical dimension (`Total`) in Elcom cube mentioned in #1395.

This PR fixes that by also parsing multiple data types defined directly in dimension metadata.

## How to test
1. Go to [this link](https://visualization-tool-git-fix-datatype-parsing-ixt1.vercel.app/en/create/new?cube=https://energy.ld.admin.ch/elcom/electricityprice&dataSource=Prod).
2. See that the cube loads correctly.
